src/FlagProvider.tsx

```diff
@@ -37,13 +39,13 @@ const FlagProvider: React.FC<React.PropsWithChildren<IFlagProvider>> = ({
   const [flagsReady, setFlagsReady] = React.useState(
     Boolean(
       unleashClient
-        ? customConfig?.bootstrap && customConfig?.bootstrapOverride !== false
+        ? (customConfig?.bootstrap && customConfig?.bootstrapOverride !== false) || unleashClient.isReady?.()
```

Comment:
Isn't this the only significant line here? Why do we need these other changes?

Reply:
It's about adding the ability to pass initialized client. Error state is also needed to make sure state kept in React SDK is matching the one from initialized client